File: gen.py
import json

raw = """
1\t9C\t3160136\tPRANEETH BHUKYA
2\t9C\t3160417\tSATHWIK REDDY GANTA
3\t9C\t3160662\tKOTHA ASHVIK
4\t9C\t3170292\tBOPPARAJU ABHIRAM
5\t9C\t3170355\tBOYINI VIVEKANANDA MUDIRAJ
6\t9C\t3170390\tLOLLA ABHIRAM
7\t9C\t3180076\tKASULA ASHWATH
8\t9C\t3180133\tSNITHIK VENGALA
9\t9C\t3180183\tNIDHISH DUMALA
10\t9C\t3180184\tYAMMANURU HARITEJA
11\t9C\t3180286\tCHARAN BATTU
12\t9C\t3190472\tATHUL M
13\t9C\t3200320\tSIDDHARTH REDDY SADIVILLA
14\t9C\t3200437\tSIDDALA RAMCHARAN
15\t9C\t3210447\tSAPAVATH JAHNAVI
16\t9C\t3210590\tK MOKSHA
17\t9C\t3230302\tBOJJA HARIKESH REDDY
18\t9C\t3230706\tEKADANTHA YADAV
19\t9C\t3240214\tNIMMAKAYALA PRATEEK REDDY
20\t9C\t3240504\tCHEEKOORI SAI CHARAN
21\t9C\t3240693\tVALLETI SAI HARSHITH
22\t9C\t3250112\tP SATHWIK REDDY
23\t9C\t3260066\tDEPA AARYAN REDDY (NA)
24\t9C\t3230719\tG MANASWINI
25\t9C\t3170068\tSHERI RITHIK REDDY
26\t9C\t3220915\tBHUKYA PRANAVI
27\t9C\t3190133\tTANABUDDI SRI BHAVESH REDDY
"""

users_str = "// ─── Generated Parent Users (Class 9-C) ───\n"
students_str = "// ─── Generated Students (Class 9-C) ───\n"

for i, line in enumerate(raw.strip().split('\n')):
    parts = line.split('\t')
    if len(parts) != 4: continue
    sno, sec, adm, name = parts
    clean_name = name.replace(" (NA)", "")
    roll = str(sno).zfill(2)
    id_num = 100 + int(sno)
    
    users_str += f"""  {{
    id: {id_num},
    username: "P{adm}A",
    password: "PARENT123",
    name: "Parent of {clean_name}",
    role: "parent",
    roleLabel: "Parent",
    department: "Parent of {clean_name} (Class 9-C)",
    phone: "+91 99999 {roll}000",
    email: "parent{adm}@example.com",
    joined: "Apr 2024",
    avatar_color: "#f57c00",
    icon: "fa-user-friends",
    childName: "{clean_name}",
    childId: "{adm}",
    childClass: "9-C",
    childRoll: "{roll}",
    notifications: []
  }},
"""
    att = 85 + (i % 15)
    gpa = round(7.5 + (i % 20) * 0.1, 1)
    students_str += f'  {{ id: "S{id_num}", admNo: "{adm}", name: "{clean_name}", class: "9-C", roll: "{roll}", gender: "Unknown", dob: "15 Jan 2011", attendance: {att}, behavior: "Good", fee_status: "Paid", gpa: {gpa}, parent: "Parent of {clean_name}" }},\n'

with open("gen_output.txt", "w", encoding="utf-8") as f:
    f.write(users_str + "\n\n" + students_str)

File: gen.js
const fs = require('fs');

const raw = `
1	9C	3160136	PRANEETH BHUKYA
2	9C	3160417	SATHWIK REDDY GANTA
3	9C	3160662	KOTHA ASHVIK
4	9C	3170292	BOPPARAJU ABHIRAM
5	9C	3170355	BOYINI VIVEKANANDA MUDIRAJ
6	9C	3170390	LOLLA ABHIRAM
7	9C	3180076	KASULA ASHWATH
8	9C	3180133	SNITHIK VENGALA
9	9C	3180183	NIDHISH DUMALA
10	9C	3180184	YAMMANURU HARITEJA
11	9C	3180286	CHARAN BATTU
12	9C	3190472	ATHUL M
13	9C	3200320	SIDDHARTH REDDY SADIVILLA
14	9C	3200437	SIDDALA RAMCHARAN
15	9C	3210447	SAPAVATH JAHNAVI
16	9C	3210590	K MOKSHA
17	9C	3230302	BOJJA HARIKESH REDDY
18	9C	3230706	EKADANTHA YADAV
19	9C	3240214	NIMMAKAYALA PRATEEK REDDY
20	9C	3240504	CHEEKOORI SAI CHARAN
21	9C	3240693	VALLETI SAI HARSHITH
22	9C	3250112	P SATHWIK REDDY
23	9C	3260066	DEPA AARYAN REDDY (NA)
24	9C	3230719	G MANASWINI
25	9C	3170068	SHERI RITHIK REDDY
26	9C	3220915	BHUKYA PRANAVI
27	9C	3190133	TANABUDDI SRI BHAVESH REDDY`;

let users = `// Generated Parent Users\n`;
let students = `// Generated Students\n`;

raw.trim().split('\n').forEach((line, i) => {
  const [sno, ssec, adm, name] = line.split('\t');
  let cleanName = name.replace(' (NA)', '');
  let id = 100 + parseInt(sno);
  users += `  {
    id: ${id},
    username: "P${adm}A",
    password: "parent123",
    name: "Parent of ${cleanName}",
    role: "parent",
    roleLabel: "Parent",
    department: "Parent of ${cleanName} (Class 9-C)",
    phone: "+91 99999 000${sno.padStart(2, '0')}",
    email: "parent${adm}@example.com",
    joined: "Apr 2024",
    avatar_color: "#f57c00",
    icon: "fa-user-friends",
    childName: "${cleanName}",
    childId: "${adm}",
    childClass: "9-C",
    childRoll: "${sno.padStart(2, '0')}",
    notifications: []
  },
`;
  students += `  { id: "S${100+parseInt(sno)}", admNo: "${adm}", name: "${cleanName}", class: "9-C", roll: "${sno.padStart(2, '0')}", gender: "Unknown", dob: "15 Jan 2011", attendance: ${85 + (i%15)}, behavior: "Good", fee_status: "Paid", gpa: ${(7.5 + (i%20)*0.1).toFixed(1)}, parent: "Parent of ${cleanName}" },\n`;
});

fs.writeFileSync('C:\\Users\\Kasula Santhosh\\OneDrive\\Desktop\\ANTIGRAVATI CAMPUSCORE\\gen_output.txt', users + '\n\n' + students);
